R/create_extendr_package.R

```diff
@@ -0,0 +1,76 @@
+
+#' Create a project for R package development with Rust
+#'
+#' @description
+#' This function creates an R project directory for package development
+#' with Rust extensions.
+#'
+#' @param path a path to new directory
+#' @param ... arguments passed on to `usethis::create_package()` and
+#' `rextendr::use_extendr()`
+#'
+#' @return Path to the newly created project or package, invisibly.
+#' @keywords internal
+#'
+#' @examples
+create_extendr_package <- function(path, ...) {
+
+  # error if usethis is not installed
+  rlang::check_installed("usethis")
+
+  args <- rlang::list2(...)
+
+  # hunch is that rstudio project text input widgets return empty strings
+  # when no value is given, want to make sure it is NULL so `use_extendr()`
+  # handles it correctly
+  args <- lapply(args, function(x) if (x == "") return(NULL) else return(x))
```

Comment:
We have `{purrr}` as dependency, so you can use that. Also, `==` will backfire horribly if `x` is `NA` or `NULL`

Reply:
I believe these widgets are based on shiny, so the text input widget should return `""` when empty. https://shiny.posit.co/r/reference/shiny/1.7.3/textinput. 

but even so, maybe we shouldn't rely on that assumption? should i do something like `if (is.na(x) | is.null(x) | nchar(x) == 0)`?

Reply:
`rlang` to the rescue!